src/H5RS.c

```diff
@@ -175,54 +317,224 @@ H5RS_wrap(const char *s)
         HGOTO_ERROR(H5E_RS, H5E_CANTALLOC, NULL, "memory allocation failed")
 
     /* Set the internal fields */
-    ret_value->s       = (char *)s;
-    ret_value->wrapped = 1;
+    ret_value->s   = (char *)s;
+    ret_value->len = HDstrlen(s);
+    ret_value->end = ret_value->s + ret_value->len;
+
+    ret_value->wrapped = TRUE;
+    ret_value->max     = 0; /* Wrapped, not allocated */
     ret_value->n       = 1;
 
 done:
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5RS_wrap() */
 
-/*--------------------------------------------------------------------------
- NAME
-    H5RS_own
- PURPOSE
-    Transfer ownership of a regular string to a reference counted string
- USAGE
-    H5RS_str_t *H5RS_own(s)
-        const char *s;          IN: String to transfer ownership of
+/*-------------------------------------------------------------------------
+ * Function:    H5RS_asprintf_cat
+ *
+ * Purpose:     This function appends formatted output to a ref-counted string,
+ *              allocating the managed string if necessary.  The formatting
+ *              string is printf() compatible.
+ *
+ * Return:      SUCCEED / FAIL
+ *
+ * Programmer:	Quincey Koziol
+ *	        Friday, September 18, 2020
+ *
+ *-------------------------------------------------------------------------
+ */
+/* Disable warning for "format not a string literal" here -QAK */
+/*
+ *      This pragma only needs to surround the sprintf() calls with
+ *      format_templ in the code below, but early (4.4.7, at least) gcc only
+ *      allows diagnostic pragmas to be toggled outside of functions.
+ */
+H5_GCC_DIAG_OFF(format - nonliteral)
```

Comment:
You have to quote "format-literal" or the formatter will mess it up, as it did here.

Reply:
Oops, I'll fix and update the PR.